internal/selfupdate/download_unix.go

```diff
@@ -0,0 +1,11 @@
+//go:build !windows
+// +build !windows
+
+package selfupdate
+
+import "os"
+
+// Remove the target binary.
+func removeResticBinary(dir, target string) error {
+	return os.Remove(target)
```

Comment:
I wonder whether we should let `os.Rename`  take care of replacing the restic binary, instead of manually deleting it first. That would make the update truly atomic.

Reply:
changed